pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -206,27 +235,25 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 		}
 		r.isAutopilotCluster = &isAutopilot
 	}
-
-	adjusted, err := adjustContainerResources(*r.isAutopilotCluster, declared, current)
+	dep, err := isDeploymentSame(*r.isAutopilotCluster, declared, unObjCurrent, reconcilerManagerAllowList, r.scheme)
```

Comment:
Usually when you have a function named with the "Is-" prefix, it just returns a bool. Can this be named better to reflect what it's returning without me needing to go look at the impl?

Reply:
Sure. When I considered about how to name it, I think the main part for this function is to decide if we have "same" declared deployment with current deployment. If not we return a data array for PATCH. So I decided to use "isDeploymentSame".  I also do not want the function name to be really long, so do you have some better suggestions?